File: JavaScript/Math-Array-String/53.js
// 2859. Sum of Values at Indices With K Set Bits


/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var sumIndicesWithKSetBits = function(nums, k) {
    // Helper function to count the number of set bits (1's) in a binary representation
    const countSetBits = (n) => {
        return n.toString(2).split('1').length - 1;
    };

    let totalSum = 0;

    // Iterate over indices and check if the count of set bits matches k
    for (let i = 0; i < nums.length; i++) {
        if (countSetBits(i) === k) {
            totalSum += nums[i];
        }
    }

    return totalSum;
};

// Example 1
const nums1 = [5, 10, 1, 5, 2];
const k1 = 1;
console.log(sumIndicesWithKSetBits(nums1, k1)); // Output: 13

// Example 2
const nums2 = [4, 3, 2, 1];
const k2 = 2;
console.log(sumIndicesWithKSetBits(nums2, k2)); // Output: 1

File: Python/Math-Array-String/53.py
# Python file


class Solution:
    def sumIndicesWithKSetBits(self, nums: List[int], k: int) -> int:
        def count_set_bits(n):
            # Count the number of set bits (1's) in the binary representation of n
            return bin(n).count('1')
        
        total_sum = 0
        
        for i, num in enumerate(nums):
            if count_set_bits(i) == k:
                total_sum += num
                
        return total_sum